src/NetTopologySuite/Operation/Union/UnionStrategy.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using NetTopologySuite.Geometries;
+
+namespace NetTopologySuite.Operation.Union
+{
+    /// <summary>
+    /// An strategy class that allows UnaryUnion to adapt to different
+    /// kinds of overlay algorithms.
+    /// </summary>
+    /// <author>Martin Davis</author>
+    public class UnionStrategy
+    {
+        private readonly Func<Geometry, Geometry, Geometry> _unionFunction;
+
+        internal UnionStrategy(Func<Geometry, Geometry, Geometry> func, bool isFloatingPrecision)
+        {
+            _unionFunction = func;
+            IsFloatingPrecision = isFloatingPrecision;
+        }
+
+        /// <summary>
+        /// Computes the union of two geometries.
+        /// This method may throw a <see cref="TopologyException"/>
+        /// if one is encountered.
+        /// </summary>
+        /// <param name="g0">A geometry</param>
+        /// <param name="g1">A geometry</param>
+        /// <returns>The union of the input</returns>
+        public Geometry Union(Geometry g0, Geometry g1) => _unionFunction(g0, g1);
```

Comment:
validate `g0` non-null before using it

also `g1`???

or just make this `internal`???

Reply:
just making it `internal`.  it still builds.